### PR TITLE
Create many-cfd-one-tunnel.md

### PR DESCRIPTION
**Pull request by abelinkinbio**
First tutorial. Let me know where questions you have.

### CODE REVIEW

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+---
+updated: 2021-05-10
+category: 🌐 Connections
+---
+
+# Run the same Tunnel across many `cloudflared` processes
+
+You can use [Cloudflare Tunnel](/connections/connect-apps) to connect applications and servers to Cloudflare's network. Tunnel relies on a piece of software, `cloudflared`, to create those connections.
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I suggest making cloudflared here a link to something, maybe github.com/cloudflare/cloudflared

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+The same Tunnel can also represent multiple, redundant instances of `cloudflared`, giving you the ability to scale instances dynamically in a replica model.
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: A tunnel doesn't represent a cloudflared, a cloudflared runs a tunnel. I suggest "The same tunnel can be run from multiple instances of cloudflared, giving you the ability to run many cloudflared replicas to scale your system when incoming traffic changes."

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+In this tutorial, we will walkthrough running an app in a Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/), and then running `cloudflared` in a separate [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I think "walkthrough" as a verb should be "walk through"

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+This architecture allows `cloudflared` instances to proxy internet traffic into whichever Kubernetes Service it was configured to. Alternatively, if your cluster already has an ingress controller (e.g. Envoy), you can use `cloudflared` to expose the ingress controller to internet traffic.
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: If this tutorial doesn't cover using cloudflared to point at an ingress controller like Envoy, then I suggest removing any reference to that idea.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+* Proxy traffic into a Kubernetes service with Tunnel
+
+**⏲️ Time to complete: 45 minutes**
+
+## Install `cloudflared`
+
+Start by [downloading and installing](/connections/connect-apps/install-and-setup/installation) the lightweight Cloudflare Tunnel daemon, `cloudflared`. On Mac, you can do so by running the following `brew` command. If you do not have Homebrew, follow the [documentation](https://docs.brew.sh/Installation) to install it.
+
+```sh 
+$ brew install cloudflare/cloudflare/cloudflared
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I wouldn't assume users are on MacOS, I'd say something generic instead, like "you can probably find cloudflared in your system's package manager, see <this page> for details"

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+## Login to to Cloudflare
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: double 'to'

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+Alternatively, you can also perform this step from the command line by running `cloudflared route dns <tunnel> <hostname>`. For example, `cloudflared tunnel route dns example-tunnel tunnel.example.com`. You can use a similar method to route traffic to `cloudflared` from a [Cloudflare Load Balancer](https://www.cloudflare.com/load-balancing/), see [docs](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/lb) for details.
+  
+## Deploy `cloudflared` 
+
+Now, we'll deploy `cloudflared` by applying its manifest. This will start a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) for running `cloudflared` and a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) with `cloudflared`'s config. When Cloudflare receives traffic for the DNS or Load Balancing hostname you configured in the previous step, it will send that traffic to the `cloudflared`'s running in this deployment. Then, those `cloudflared` instances will proxy the request to your app's Service.
```
Comment (adamchalmers) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: "What manifest? This tutorial never mentioned a manifest. Where do I download the manifest from?"
I suggest linking to it, or copying it inline here.

You'll also need to link or explain the example service that's being proxied to. In my example I used "httpbin" and supplied a manifest for it.

**Review by adamchalmers — CHANGES_REQUESTED** [review]

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 1 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 2 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: Fair point - could have led to confusion. ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 3 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 4 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✂️

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 5 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 6 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✂️

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 7 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I was at a bit of a loss for linking the manifest here. I may simply copy inline. I'll look into it tonight.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
+---
+updated: 2021-05-10
```
Comment (TownLake) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: Edit for Tuesday to 2021-05-11

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+Alternatively, you can also perform this step from the command line by running `cloudflared route dns <tunnel> <hostname>`. For example, `cloudflared tunnel route dns example-tunnel tunnel.example.com`. You can use a similar method to route traffic to `cloudflared` from a [Cloudflare Load Balancer](https://www.cloudflare.com/load-balancing/), see [docs](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/lb) for details.
+  
+## Deploy `cloudflared` 
+
+Now, we'll deploy `cloudflared` by applying its manifest. This will start a [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) for running `cloudflared` and a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) with `cloudflared`'s config. When Cloudflare receives traffic for the DNS or Load Balancing hostname you configured in the previous step, it will send that traffic to the `cloudflared`'s running in this deployment. Then, those `cloudflared` instances will proxy the request to your app's Service.
```
Comment (TownLake) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: probably worth walking through the architecture at a high-level here - how the instances scale up or down and describing once again how the tunnel can map to one or many, each instance will use the same credentials file etc

**Review by TownLake — CHANGES_REQUESTED** [review]

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
+---
+updated: 2021-05-10
+category: 🌐 Connections
+---
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md:5]: Do you want to add a difficulty here? Most other tutorials seem to indicate that.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+# Run the same Tunnel across many `cloudflared` processes
+
+You can use [Cloudflare Tunnel](/connections/connect-apps) to connect applications and servers to Cloudflare's network. Tunnel relies on a piece of software, [cloudflared](https://github.com/cloudflare/cloudflared), to create those connections.
+
+The same tunnel can be run from multiple instances of `cloudflared`, giving you the ability to run many `cloudflared` replicas to scale your system when incoming traffic changes.
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: The title says `Tunnel` but here you use `tunnel`. Let's aim for consistency.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+In this tutorial, we will walk through running an app in a Kubernetes [Service](https://kubernetes.io/docs/concepts/services-networking/service/), and then running `cloudflared` in a separate [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: > an app

feels overly informal, I'd use "application"

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+```sh
+$ cloudflared tunnel create example-tunnel
+INFO[2020-09-05T10:48:34+01:00] Writing tunnel credentials to /Users/cf000197/.cloudflared/ef824aef-7557-4b41-a398-4684585177ad.json. cloudflared chose this file based on where your origin certificate was found.
+INFO[2020-09-05T10:48:34+01:00] Keep this file secret. To revoke these credentials, delete the tunnel.
+INFO[2020-09-05T10:48:34+01:00] Created tunnel example-tunnel with id ef824aef-7557-4b41-a398-4684585177ad
+```
+
+## Upload the Tunnel credentials file to Kubernetes 
+
+Next, you will need to upload the credential file which was generated to Kubernetes as a secret. You will also need to provide the filepath that the Tunnel credentials file was created under as well. You can find that path in the output of `cloudflared tunnel create <example-tunnel>` above.
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: > Next, you will need to upload the credential file which was generated to Kubernetes as a secret.

This reads strange.
I'd write it as:  "Next, you will upload the generated Tunnel credential file as a secret to your Kubernetes cluster".

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 7 above)
Comment (nmldiegues) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: IMO we should heavily delegate to https://github.com/cloudflare/argo-tunnel-examples/tree/master/named-tunnel-k8s

I'd avoid duplicating effort/contents

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    2021-05-04T17:39:49Z INF Settings: map[config:/etc/cloudflared/config/config.yaml cred-file:/etc/cloudflared/creds/credentials.json credentials-file:/etc/cloudflared/creds/credentials.json metrics:0.0.0.0:2000 no-autoupdate:true]
+    2021-05-04T17:39:49Z INF Generated Connector ID: 4c5dc5d3-8e10-480e-ac74-e385e591553e
+    2021-05-04T17:39:49Z INF Initial protocol h2mux
+    2021-05-04T17:39:49Z INF Starting metrics server on [::]:2000/metrics
+    2021-05-04T17:39:49Z INF Connection 1daced2f-466c-4610-8ba6-7642a8ddec68 registered connIndex=0 location=MCI
+    2021-05-04T17:39:50Z INF Connection 1a5276bc-3313-4bb7-a677-d93deccab24f registered connIndex=1 location=DFW
+    2021-05-04T17:39:51Z INF Connection aa7adacc-e855-4b11-bf41-e113419b7ef4 registered connIndex=2 location=MCI
+    2021-05-04T17:39:51Z INF Connection a8055c76-2a90-4be5-8dc9-ebaa5c58fb5f registered connIndex=3 location=DFW
+```
+
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md:99]: Something cool to do at this point would be to manually change the number of replicas for the deployment.

```
kubectl scale --replicas=3 rs/cloudflared
```

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+## Visit your hostname
+
+And you'll see the httpbin welcome page. 
+
+We love to hear your feedback! Join a discussion with other community members at https://community.cloudflare.com/c/performance/argo-tunnel
```
Comment (nmldiegues) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I think we should have a "Wrap up" section that summarizes:
 - The same Tunnel can be ran in many cloudflared processes
 - This allows for elastic scaling
 - But can also be useful for graceful rolling upgrades of cloudfared (changing their binary and/or configuration)

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 12 above)
Comment (chungthuang) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I would say it's running an app as a Kubernetes Service.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+## Install `cloudflared`
+
+Start by [downloading and installing](/connections/connect-apps/install-and-setup/installation) the lightweight Cloudflare Tunnel daemon, `cloudflared`. Reference our installation guide for instructions on how to install `cloudflared` on your operating system. 
+
+## Login to Cloudflare
+
+Once installed, you can use the `tunnel login` command in `cloudflared` to obtain a certificate.
+
+```sh
+$ cloudflare tunnel login
```
Comment (chungthuang) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: cloudflared.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+## Create your Tunnel
+
+In the example below, simply change <example-tunnel> to the name you wish to assign to your tunnel.
+
+```sh
+$ cloudflared tunnel create example-tunnel
```
Comment (chungthuang) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md:41]: Latest cloudflared output looks like this
```
$ cloudflared tunnel create demo
Tunnel credentials written to /Users/cf000197/.cloudflared/416a0fd5-b771-4dda-9949-966cc3652c94.json. cloudflared chose this file based on where your origin certificate was found. Keep this file secret. To revoke these credentials, delete the tunnel.

Created tunnel demo with id 416a0fd5-b771-4dda-9949-966cc3652c94
```

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+
+Once installed, you can use the `tunnel login` command in `cloudflared` to obtain a certificate.
+
+```sh
+$ cloudflare tunnel login
+```
+
+## Create your Tunnel
+
+In the example below, simply change <example-tunnel> to the name you wish to assign to your tunnel.
```
Comment (abracchi-tw) [inline products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]:
```suggestion
In the example below, simply change `<example-tunnel>` to the name you wish to assign to your Tunnel.
```

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 11 above)
Comment (abracchi-tw) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: let's stick to `Tunnel`, I think that's what we went for in most other tutorials.

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 10 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md:5]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 11 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 12 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: ✅

File: products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md (hunk 7 above)
Comment (abelinkinbio) [reply products/cloudflare-one/src/content/tutorials/many-cfd-one-tunnel.md]: I've linked out to the named-tunnels-k8s tutorial for the cloudflared.yaml and app.yaml references. I agree we should try to have consistency amongst the two, but also want to ensure we're getting full coverage.

**Review by abracchi-tw — APPROVED** [review]

**Review by TownLake — APPROVED** [review]

**Review by adamchalmers — APPROVED** [review]

**Review by YoutacRSVARs — COMMENTED** [review]
__

**Pull request merged**